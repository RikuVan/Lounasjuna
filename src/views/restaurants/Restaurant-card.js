import React, {PropTypes} from 'react'
import './Restaurants.css'
import StarRating from '../../components/Star-rating'
import Button from '../../components/Button'
//import RestaurantVotes from './Restaurant-votes'

//Helper to add users to RestaurantVotes
//const mapToUsers = (votes = [], users = {}) =>
//votes.map(id => users[id].displayName)

/**
 * SPRINT 3
 * TODO: This panel takes a list of users in RestaurantsVote component
 * Change where the button is active depending on whether the user has voted for the restaurant
 * A vote function should be passed as prop to this dumb component to be used as a callback
 *
 */

export const RestaurantCard = (
  {
    name,
    address,
    link,
    rating,
    type,
  },
) => {
  return (
    <article className="Restaurant">
      <h2 className="Restaurant-name">
        {link &&
          <a href={link} className="Restaurant-www" target="_blank">
            <span className="fa fa-home" />
          </a>}
        {name}
      </h2>
      <div className="Restaurant-content">
        <StarRating rating={rating ? Math.round(rating) : 0} />
        <p>{type}</p>
        <p>{address}</p>
        {/*<RestaurantVotes votes={mapToUsers(votes, users)} />} */}
      </div>
      <div className="Restaurant-footer">
        <Button disabled={false} onClick={() => {}}>
          LÃ¤hden junaan mukaan
        </Button>
      </div>
    </article>
  )
}

RestaurantCard.propTypes = {
  name: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
  link: PropTypes.string,
  rating: PropTypes.string,
  type: PropTypes.string,
}

export default RestaurantCard
