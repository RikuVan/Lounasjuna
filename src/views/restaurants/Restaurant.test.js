import React from 'react'
import {RestaurantCard} from './Restaurant-card'
import {shallow, mount} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

/**
 * Example component tests
 */

test('Restaurant contains title and link', () => {
  //this is to avoid logging of propType errors
  console.error = jest.genMockFn()
  const handleSelect = jest.fn()
  const data = {
    name: 'Restaurant',
    address: '10 some street',
    rating: 2,
    link: 'http://chineseplace.com',
    type: 'Chinese',
    rating: 1,
    userId: '1',
    uid: '2',
    votes: ['1'],
    users: {
      1: {displayName: 'some dude'},
    },
    currentVote: 1,
    handleSelect,
  }
  const title = (
    <h2 className="Restaurant-name">
      <a href="http://chineseplace.com" className="Restaurant-www">
        <span className="fa fa-home" />
      </a>Restaurant
    </h2>
  )
  const component = mount(<RestaurantCard {...data} />)
  component.find('button').simulate('click')
  expect(component.contains(title)).toBe(true)
  expect(handleSelect).toHaveBeenCalledTimes(1)
  expect(handleSelect).toHaveBeenCalledWith('1', '2')
})

test('Restaurant card snapshot test', () => {
  console.error = jest.genMockFn()
  const component = shallow(<RestaurantCard />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
