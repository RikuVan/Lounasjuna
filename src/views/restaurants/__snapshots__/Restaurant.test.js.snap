exports[`test Restaurant card snapshot test 1`] = `
<article
  className="Restaurant">
  <h2
    className="Restaurant-name" />
  <div
    className="Restaurant-content">
    <StarRating
      rating={0} />
    <p />
    <p />
  </div>
</article>
`;
